### PR TITLE
fix: remove setup files from coverage

### PR DESCRIPTION
**Pull request by g4rry420**
Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

Ref: https://github.com/vitest-dev/vitest/pull/2489
Closes: #2190

### CODE REVIEW

File: packages/vitest/src/node/config.ts [hunk 1]
```diff
@@ -190,6 +190,7 @@ export function resolveConfig(
         ?? resolve(resolved.root, file),
     ),
   )
+  resolved.coverage.exclude.push(...resolved.setupFiles.map(file => file.replace(`${resolved.root}/`, '')))
```
Comment (sheremet-va) [inline packages/vitest/src/node/config.ts:193]: Maybe it's better to use `relative` from `pathe` module?

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Please, add an actual test case to `test/coverage`. Current test is pretty meaningless.

**Comment by g4rry420** [comment]
Thank you for suggesting the use of `relative` from `pathe` module. I have added that.
For testing, honestly, I am not really good with it. Can you please suggest me what should I add in it ?

**Comment by sheremet-va** [comment]
> Thank you for suggesting the use of `relative` from `pathe` module. I have added that.
> For testing, honestly, I am not really good with it. Can you please suggest me what should I add in it ?

There is `test/coverage-test` folder. You need to add a setup file and check that it's not in the report.

**Review by sheremet-va — CHANGES_REQUESTED** [review]
I don't see a check inside coverage test that setup file is not in a report?

**Comment by g4rry420** [comment]
I have added the test. Please, let me know if you see any issues :)

**Comment by g4rry420** [comment]
One thing, I have noticed is that, if for some reason, a user intentionally wants to include the `setup` file in the coverage, then they can't do that (even if they mention that file in the `coverage.include` option). I believe, this is expected.

**Review by sheremet-va — APPROVED** [review]

**Review by AriPerkkio — CHANGES_REQUESTED** [review]
Currently the test case does not fail if the implementation is commented out. Let's add some more checks there:

- Keep the current `setup.ts` and its test assertions as is
- Create `test/coverage-test/src/another-setup.ts`
- Use `src/another-setup.ts` as second setupFile in configuration
- Add assertions to the test case to check that it does not end up in the coverage report folder

Final test case should assert that both setupFiles are not present in coverage report.

**Review by AriPerkkio — APPROVED** [review]

**Review by sheremet-va — APPROVED** [review]
@AriPerkkio should we add a test that file imported in setupFile is still inside coverage report? This is expected behaviour, right?

```ts
// ./utils.ts
export function defineGlobal() {
  globalThis.testing = true
}
```

```ts
// ./vitest.setup.ts
import { defineGlobal } from './utils.ts'
defineGlobal()
```

Setup file should not be in the report, but utils should?

**Review by sheremet-va — CHANGES_REQUESTED** [review]
^ See comment above

**Comment by AriPerkkio** [comment]
> This is expected behaviour, right?

The imported files will end up in coverage report and we should not try to change that - it would be quite complex. I don't think adding test case for validating it is that important.

The current changes look good to me, and tests are covering the functionality well.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**